File: weather-mvp-dagger/src/main/java/com/baronzhang/android/weather/view/selectcity/SelectCityModule.java
package com.baronzhang.android.weather.view.selectcity;

import dagger.Module;
import dagger.Provides;

import com.baronzhang.android.weather.di.ActivityScoped;
import com.baronzhang.android.weather.view.contract.SelectCityContract;

/**
 * @author baronzhang (baron[dot]zhanglei[at]gmail[dot]com)
 *         2016/11/30
 */
@Module
public class SelectCityModule {
    private SelectCityContract.View view;

    public SelectCityModule(SelectCityContract.View view) {
        this.view = view;
    }

    @Provides
    @ActivityScoped
    SelectCityContract.View provideSelectCityContractView() {
        return view;
    }
}

File: weather-mvp-dagger-kotlin/src/main/java/com/minimalist/weather/kotlin/view/selectcity/SelectCityModule.kt
package com.minimalist.weather.kotlin.view.selectcity

import com.minimalist.weather.kotlin.di.ActivityScoped
import com.minimalist.weather.kotlin.di.FragmentScoped
import com.minimalist.weather.kotlin.view.contract.SelectCityContract
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class SelectCityModule {

    @FragmentScoped
    @ContributesAndroidInjector
    internal abstract fun selectCityPresenter(): SelectCityFragment

    @ActivityScoped
    @Binds
    abstract fun selectCityPresenter(presenter: SelectCityPresenter): SelectCityContract.Presenter
}
